pkg/controller/backup_controller.go

```diff
@@ -833,15 +833,6 @@ func persistBackup(backup *pkgbackup.Request,
 		persistErrs = append(persistErrs, errs...)
 	}
 
-	csiSnapshotJSON, errs := encode.ToJSONGzip(csiVolumeSnapshots, "csi volume snapshots list")
-	if errs != nil {
-		persistErrs = append(persistErrs, errs...)
-	}
-
-	csiSnapshotContentsJSON, errs := encode.ToJSONGzip(csiVolumeSnapshotContents, "csi volume snapshot contents list")
-	if errs != nil {
-		persistErrs = append(persistErrs, errs...)
-	}
 	csiSnapshotClassesJSON, errs := encode.ToJSONGzip(csiVolumeSnapshotClasses, "csi volume snapshot classes list")
```

Comment:
Can you elaborate why we are leaving class? 
Any flow which relies on it?

Reply:
The VolumeSnapshotClass is left for the CSI cross-cluster restore.
The VolumeSnapshotClasses are synced to the new cluster by the backup sync controller, and they are used for backup deletion on the new cluster.
If the VolumeSnapshotClasses are not synced, there is not a good chance to create them for VolumeSnapshot deletion of the backup.

Reply:
do we really need to do this as part of Backup Sync? 
OR can we do it inline during GC?

Reply:
I haven't thought that through yet, but my feeling is that moving the VolumeSnapshotContent to GC may change the workflow significantly.

I think this issue was caused by two factors:
* VolumeSnapshotContent deletion depending on the VolumeSnapshotClass, which is not a good design. PersistentVolume deletion doesn't depend on StorageClass.
* Backup deletion doesn't have a way to create the needed resources before deleting something.

Reply:
After some testing, I found VSC deletion and creation can work without VolumeSnapshotClass.
I will verify whether Velero workflow can work without syncing the VolumeSnapshotClass.